jujugui/static/gui/src/app/app.js

```diff
@@ -1817,6 +1817,8 @@ YUI.add('juju-gui', function(Y) {
           this._clearDeployment.bind(this)]
       ]);
 
+      // Listen for change state events from the canvas components.
+      this.on('*:changeState', this._changeState, this);
```

Comment:
Lets remove this and the `_changeState` method entirely and instead rely on passing the method around. This way we remove another piece of our dependence on YUI.